plugins/sigma.layout.fruchtermanReingold/sigma.layout.fruchtermanReingold.js

```diff
@@ -0,0 +1,285 @@
+
+        if (dist > 0) {
+          var limitedDist = Math.min(maxDisplace * self.config.speed, dist);
+          n.fr_x += xDist / dist * limitedDist;
+          n.fr_y += yDist / dist * limitedDist;
+        }
+      }
+    }
+
+    return this.running;
+  };
+
+  FruchtermanReingold.prototype.go = function () {
+    this.iterCount = this.config.maxIterations;
+
+    while (this.running) {
+      this.atomicGo();
+    };
+
+    this.stop();
+  };
+
+  FruchtermanReingold.prototype.start = function() {
+    if (this.running) return;
+
+    var nodes = this.graph.nodes();
+
+    this.running = true;
+
+    if (!this.started) {
+      // Init nodes
+      for (var i = 0; i < nodes.length; i++) {
+        nodes[i].fr_x = nodes[i].x;
+        nodes[i].fr_y = nodes[i].y;
+        nodes[i].fr = {
+          dx: 0,
+          dy: 0
+        };
+      }
+      this.started = true;
+      eventEmitter.dispatchEvent('start');
+      this.go();
+    }
+  };
+
+  FruchtermanReingold.prototype.stop = function() {
+    var self = this,
+        nodes = this.graph.nodes();
+
+    this.running = false;
+    eventEmitter.dispatchEvent('stop');
+
+    if (this.easing) {
+      eventEmitter.dispatchEvent('interpolate');
+      sigma.plugins.animate(
+        self.sigInst,
+        {
+          x: 'fr_x',
+          y: 'fr_y'
+        },
+        {
+          easing: self.easing,
+          onComplete: function() {
+            self.sigInst.refresh();
+            for (var i = 0; i < nodes.length; i++) {
+              nodes[i].fr = null;
+              nodes[i].fr_x = null;
+              nodes[i].fr_y = null;
+            }
+            eventEmitter.dispatchEvent('stop');
+          },
+          duration: self.duration
+        }
+      );
+    }
+    else {
+      // Apply changes
+      for (var i = 0; i < nodes.length; i++) {
+        nodes[i].x = nodes[i].fr_x;
+        nodes[i].y = nodes[i].fr_y;
+      }
+
+      this.sigInst.refresh();
+
+      for (var i = 0; i < nodes.length; i++) {
+        nodes[i].fr = null;
+        nodes[i].fr_x = null;
+        nodes[i].fr_y = null;
+      }
+    }
+  };
+
+  /**
+   * Interface
+   * ----------
+   */
+  var instance = null;
+
+  sigma.layouts.fruchtermanReingold.configure = function(sigInst, config) {
+
+    // Create fruchterman if undefined
+    if (!instance) {
+      instance = new FruchtermanReingold();
+    }
+
+    instance.init(sigInst, config);
+
+    return eventEmitter;
+  };
+
+  sigma.layouts.fruchtermanReingold.start = function(sigInst, config) {
+
+    if (sigInst) {
+      this.configure(sigInst, config);
+    }
+
+    instance.start();
+
+    return eventEmitter;
+  };
+
+  sigma.layouts.fruchtermanReingold.isRunning = function() {
+    return !!instance && instance.running;
+  };
+
+  sigma.layouts.fruchtermanReingold.progress = function() {
+    return (instance.config.maxIterations - instance.iterCount) / instance.config.maxIterations;
```

Comment:
Does the algorithm always perform the same number of iterations ? if so, it is not only the `maxIterations`, it is the number of iterations.